zipline/lib/_windowtemplate.pxi

```diff
@@ -84,7 +86,9 @@ cdef class AdjustedArrayWindow:
         # Apply any adjustments that occured before our current anchor.
         # Equivalently, apply any adjustments known **on or before** the date
         # for which we're calculating a window.
-        while self.next_adj < anchor:
+        while (self.next_adj != -1
```

Comment:
(I think doing that would also remove the need to return -1 from `pop_next_adj` above.)